core/io/resource.cpp

```diff
@@ -151,15 +151,20 @@ bool Resource::editor_can_reload_from_file() {
 void Resource::reset_state() {
 }
 Error Resource::copy_from(const Ref<Resource> &p_resource) {
-	ERR_FAIL_COND_V(p_resource.is_null(), ERR_INVALID_PARAMETER);
-	if (get_class() != p_resource->get_class()) {
+	HashMap<Ref<Resource>, Ref<Resource>> tmp;
+	return reset_data_from(p_resource, tmp, false);
+}
+
+Error Resource::reset_data_from(const Ref<Resource> &p_from, HashMap<Ref<Resource>, Ref<Resource>> &remap_cache, const bool p_keep_local_to_scene_resource) {
```

Comment:
Let me suggest that instead of adding a new function `copy_from()` has new arguments added with defaults. The declaration in the header would be this:
```
Error copy_from(const Ref<Resource> &p_from, HashMap<Ref<Resource>, Ref<Resource>> &p_remap_cache = {}, bool p_keep_local_to_scene_resource = false);
```